src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -478,61 +479,74 @@ def build_container_from_source_with_buildpack(self, image_name, source):  # pyl
 
-        logger.debug(f"Calling '{' '.join(command)}'")
-        try:
-            is_non_supported_platform = False
-            with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
+        could_build_image = False
+        for builder_image in builder_image_list:
+            # Run 'pack build' to produce a runnable application image for the Container App
+            # Specify the image as the 'build-cache' image to ensure that the local cache is used for build layers
+            command = [pack_exec_path, 'build', cache_image_name, '--builder', builder_image, '--path', source, '--tag', image_name]
 
-                # Stream output of 'pack build' to warning stream
-                while process.stdout.readable():
-                    line = process.stdout.readline()
-                    if not line:
-                        break
+            logger.debug(f"Calling '{' '.join(command)}'")
+            try:
+                is_non_supported_platform = False
+                is_non_supported_os = False
+                with subprocess.Popen(command, stdout=subprocess.PIPE) as process:
 
-                    stdout_line = str(line.strip(), 'utf-8')
-                    logger.warning(stdout_line)
-                    if not is_non_supported_platform and "No buildpack groups passed detection" in stdout_line:
-                        is_non_supported_platform = True
+                    # Stream output of 'pack build' to warning stream
+                    while process.stdout.readable():
+                        line = process.stdout.readline()
+                        if not line:
+                            break
 
-                # Update the result of process.returncode
-                process.communicate()
-                if is_non_supported_platform:
-                    raise ValidationError("Current buildpacks do not support the platform targeted in the provided source code.")
+                        stdout_line = str(line.strip(), 'utf-8')
+                        logger.warning(stdout_line)
 
-                if process.returncode != 0:
-                    raise CLIError("Non-zero exit code returned from 'pack build'; please check the above output for more details.")
+                        # Check if the application is targeting a platform that's found in the current builder
+                        if not is_non_supported_platform and "No buildpack groups passed detection" in stdout_line:
+                            is_non_supported_platform = True
 
-                logger.debug(f"Successfully built image {image_name} using buildpacks.")
-        except ValidationError as ex:
-            raise ex
-        except Exception as ex:
-            raise CLIError(f"Unable to run 'pack build' command to produce runnable application image: {ex}") from ex
+                        # Check if the application has a version that it can target from the current builder
```

Comment:
Nit: please consider updating this comment to talk about the runtime images and all. Maybe it's just me, but this comment as of today feels very similar to the other `Check if the application is targeting a platform that's found in the current builder` above